### PR TITLE
Add bare init option

### LINKED ISSUE #236: Remove "anomalous_usage_of_7zip.yml" from `contentctl init`
Hi, is there a way with the CLI or manually using Python to remove the "anomalous_usage_of_7zip.yml" file that is created into `templates/detections/endpoint` when we do `contentctl init`?

### PR DESCRIPTION
**Pull request by 0xC0FFEEEE**
Resolves #236

### CODE REVIEW

File: contentctl/actions/initialize.py [hunk 1]
```diff
@@ -17,29 +17,37 @@ def execute(self, config: test) -> None:
         
         YmlWriter.writeYmlFile(str(config.path/'contentctl.yml'), config.model_dump()) 
 
-        #Create the following empty directories:
-        for emptyDir in ['lookups', 'baselines', 'docs', 'reporting', 'investigations']:
-            #Throw an error if this directory already exists
-            (config.path/emptyDir).mkdir(exist_ok=False)
+        if config.bare:
+            #Create the following empty directories:
+            for emptyDir in ['lookups', 'baselines', 'docs', 'reporting', 'investigations', 'app_template',
```
[the remark this answers is not in the document]
Comment (0xC0FFEEEE) [reply contentctl/actions/initialize.py]: Agreed! To be honest I was loosly aware of the content of the app_template directory but wan't entirely sure if it was required or not.

**Comment by 0xC0FFEEEE** [comment]
Should deployments be included in the bare option as well? I'm reasonably sure they should.

That said, I can't actually figure out how to define and reference a new deployment type and reference it in a detection atm!

**Comment by pyth0n1c** [comment]
> Should deployments be included in the bare option as well? I'm reasonably sure they should.
> 
> That said, I can't actually figure out how to define a new deployment type and reference it in a detection atm!

This is... complicated and has tons of room for improvement.  The confusion comes from the legacy history of the tool.
Right now, the `type` field of a detection:
Which is a String Enum:
https://github.com/splunk/contentctl/blob/b498bd768e4209fc34675a7c9733298b1c1a6d7b/contentctl/objects/enums.py#L6-L18

Is used to determine what the default Deployment is (which contains information about RBA, scheduling, whether this type is exempted from automated testing, and other things)
https://github.com/splunk/contentctl/blob/b498bd768e4209fc34675a7c9733298b1c1a6d7b/contentctl/objects/deployment.py#L30-L47

However, the deployment section can actually be defined in the Detection itself to override those defaults in the "default" deployments:
https://github.com/splunk/contentctl/blob/b498bd768e4209fc34675a7c9733298b1c1a6d7b/contentctl/objects/deployment.py#L34

We do this for SOME Baselines, as they often REALLY need a highly custom schedule, like once a week, but not for any non-baseline detections:
https://github.com/splunk/security_content/blob/a20338db1fd40348d4cc69a2105e8126d63a39b3/baselines/baseline_of_cloud_infrastructure_api_calls_per_user.yml#L48-L53


Because types are hardcoded in that enum, I don't think you can define your own Deployment File and actually _use it_ in a Detection without updating the Enum. Which is NOT how it should be.

We should do the following, probably:
1. Get rid of the enum for type:, replacing it with a dynamically constructed list of objects parsed from `deployments/`
2. Update the Deployment object with a bit of extra fields/logic to express things like "does this need to be tested" so that all the meaning is conveyed in the YML and not hardcoded into the logic of the contentctl tool.
3. Consider renaming the `type:` field in a detection to `deployment:` and having its typing be `deployment: Deployment = Field(...)` or something similar. We should still allow inline declaration of a deployment, as we do today, for one-off custom behavior in a detection.  We already have a Deployment field, so we probably don't need both a `type:` and `deployment:` field anyway.
4. Update all our content YMLs (and the rest of the contentctl code) to use the new structure of Deployment and Detection objects.


This is a lot of work, but think it makes total sense and further treats "everything as content" which has been the goal all along :) 
Thoughts?

**Comment by 0xC0FFEEEE** [comment]
@pyth0n1c Thanks for such a detailed response!

I think the proposed changes make sense. Appreciate that will be quite a lot of work, but would definitely be worth the effort :)

In the meantime I'll update this PR to include the default baselines.

**Comment by pyth0n1c** [comment]
> @pyth0n1c Thanks for such a detailed response!
> 
> I think the proposed changes make sense. Appreciate that will be quite a lot of work, but would definitely be worth the effort :)
> 
> In the meantime I'll update this PR to include the default baselines.

Hey @0xC0FFEEEE !
In an effort to get this merged sooner rather than later, I have updated the PR a bit to
1.  copy the deployments folder (even when --bare is used) as discussed.
2.  Add some more comments/documentation/explanation to the CLI and initialize.py
3.  Make sure that README and app_template are always created, even if --bare is used, and dedup the code for it.

Here is the link that that PR which I am merging in just a few moments:
https://github.com/splunk/contentctl/pull/288

As such, I will close this one out, but the functionality this added will be in the next release shortly (and in the `develop` branch right away)~

**Pull request closed without merging**